### ISSUE TITLE
Parsing error: Unexpected token =>

### ISSUE THREAD

**Issue by creeperyang** [eslint/eslint#4740]
code: 

```js
const task = (fn) => async () => {
    const start = new Date();
    console.log(`[${format(start)}] Starting '${fn.name}'...`);
    await fn();
    const end = new Date();
    const time = end.getTime() - start.getTime();
    console.log(`[${format(end)}] Finished '${fn.name}' after ${time}ms`);
};
```

error:

```bash
error  Parsing error: Unexpected token =>
```

config:

**airbnb/base**  (`eslint-config-airbnb@2.1.1, eslint@1.10.3`)

**Comment by eslintbot** [comment]
@creeperyang Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by gyandeeps** [comment]
We do not support experimental ES7 feature right now. In your case you have `async` and `await`.
If you what to support those then you can use [babel-eslint](https://github.com/babel/babel-eslint)

**Comment by creeperyang** [comment]
Thanks.

**Comment by vinogradov** [comment]
You don't have to use `babel-eslint` anymore since eslint supports es7/es8.

**Comment by dwang68** [comment]
i am still seeing this issue on v3.17.1, here is my eslint config:
 `{
    "env": {
        "browser": true,
        "node": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "sourceType": "module",
        "ecmaVersion": 7
    },
    "rules": {
        "indent": [
            "error",
            "tab"
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "single"
        ],
        "semi": [
            "error",
            "always"
        ]
    }
}`

**Comment by vinogradov** [comment]
@dwang68 I have `'ecmaVersion': 8` and it works. So first I would suggest to set `8` and see.

**Comment by dwang68** [comment]
@vinogradov yeah, that works perfectly! i think it's because of async function. i saw another thread for async/await that also suggests  setting it to 8.

**Comment by ilyavolodin** [comment]
`async`/`await` are part of es2017 spec (i.e. `ecmaVersion: 8`). es2016 only added one syntax change to the language - `**` operator.
